CHANGELOG.md

```diff
@@ -1,5 +1,42 @@
 # Changelog
 
+## v0.31.4
+
+*April 12th, 2019*
+
+This release fixes a regression from v0.31.3 which used the peer's `SocketAddr` to add the peer to
+the address book. This swallowed the peer's self-reported port which is important in case of reconnect.
+It brings back `NetAddress()` to `NodeInfo` and uses it instead of `SocketAddr` for adding peers.
+Additionally, it improves response time on the `/validators` or `/status` RPC endpoints.
+As a side-effect it makes these RPC endpoint more difficult to DoS and fixes a performance degradation in `ExecCommitBlock`.
+
+Special thanks to external contributors on this release:
+@brapse, @guagualvcha, @mydring
+
+### BREAKING CHANGES:
+
+* Go API
+  - [p2p] [\#3545](https://github.com/tendermint/tendermint/pull/3545) The `AddrBook` interface method `MarkAsGood` now only takes a `p2p.ID` instead of a `p2p.NetAddress`
+  - [p2p] [\#3011](https://github.com/tendermint/tendermint/pull/3011) Remove `ListOfKnownAddresses` from the `AddrBook` interface
+
+### IMPROVEMENTS:
+- [p2p] [\#3463](https://github.com/tendermint/tendermint/pull/3463) Do not log "Can't add peer's address to addrbook" error for a private peer
+- [p2p] [\#3547](https://github.com/tendermint/tendermint/pull/3547) Fix a couple of annoying typos (@mdyring)
```

Comment:
should we even include such minor things?

Reply:
I'm not entirely sure where to draw the line here. At least for external contributors, I think we should include their contributions even if minor.

Reply:
If the changelog is a channel for communication with peers who contribute or use the codebase we should announce all the things, including ADRs, maybe it sparks more collaboration on those.

Reply:
https://github.com/tendermint/tendermint/pull/3550